src/transformers/benchmark/benchmark_utils.py

```diff
@@ -307,6 +321,16 @@ def stop_memory_tracing(
                     frame=frame, cpu=Memory(cpu_mem_inc), gpu=Memory(gpu_mem_inc), cpu_gpu=Memory(cpu_gpu_mem_inc),
                 )
             )
+
```

Comment:
We add a current memory, which does not measure how much a line has added to the memory but how much memory is needed in total by this line. @thomwolf - I noticed that on GPU the highest `py3nvml` according to this code actually does not give the same results as `nvidia-smi`. I think the reason is that in order to measure the consumption per line we constantly reinitialize the memory tracing with `py3nvml.nvmlInit()` and `py3nvml.nvmlShutdown()` so that we don't trace added memory + currently hold memory. Also I think we would have to trace not only `transformers` code line by line but also all `PyTorch` code in order to trace the real memory peak usage. This is to slow though. Therefore I decided in order to measure the peak memory (which is usually the most important metric) to use PyTorch's `torch.cuda.max_memory_reserved` method which correspondns 1-to-1 to what is shown in `nvidia-smi`.